Description: Fix the compatibility with the Servlet API 3.1
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/main/java/org/apache/velocity/tools/view/ParameterTool.java
+++ b/src/main/java/org/apache/velocity/tools/view/ParameterTool.java
@@ -195,13 +195,13 @@
                 Array.getLength(value) == 1);
     }
 
-    private Map<String,Object> expandSingletonArrays(Map<String,Object> original)
+    private Map<String,Object> expandSingletonArrays(Map<String,String[]> original)
     {
         Map<String,Object> expanded = new HashMap<String,Object>(original);
         for (Map.Entry<String,Object> entry : expanded.entrySet())
         {
-            Object value = entry.getValue();
-            if (isSingletonArray(value))
+            String[] value = (String[]) entry.getValue();
+            if (value != null && value.length == 1)
             {
                 entry.setValue(Array.get(value, 0));
             }
--- a/src/main/java/org/apache/velocity/tools/view/ImportSupport.java
+++ b/src/main/java/org/apache/velocity/tools/view/ImportSupport.java
@@ -510,6 +510,12 @@
                     {
                         bos.write(b);
                     }
+
+                    public boolean isReady() {
+                        return true;
+                    }
+
+                    public void setWriteListener(javax.servlet.WriteListener listener) { }
                 };
             return sos;
         }
